### PR TITLE
Add function GetSelectedStringFromItem for SelectedString serialization of DropDown Node

### PR DESCRIPTION
**Pull request by QilongTang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

This is a follow up PR of https://github.com/DynamoDS/Dynamo/pull/9367, fixing the comparison logic so selection can be maintained during Populating Drop Down Items again.

@AndyDu1985 @ZiyunShang I have sent another PR https://github.com/DynamoDS/DynamoRevit/pull/2324 in DynamoRevit to correct the behavior for categories node, below are the default behaviors for the Revit DropDowns as examples

```Json
[
    {
      "ConcreteType": "DSRevitNodesUI.Levels, DSRevitNodesUI",
      "SelectedIndex": 2,
      "SelectedString": "Level 1",
      "NodeType": "ExtensionNode",
      "Id": "2a63d7b92be24b4c9570c3c2eb3c96f1",
      "Inputs": [],
      "Outputs": [
        {
          "Id": "5ae25daee90f4373bc41b893f6808e62",
          "Name": "Levels",
          "Description": "The selected Levels",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Disabled",
      "Description": "Select a level in the active document"
    },
    {
      "ConcreteType": "DSRevitNodesUI.Categories, DSRevitNodesUI",
      "SelectedIndex": 0,
      "SelectedString": "Adaptive Points",
      "NodeType": "ExtensionNode",
      "Id": "0442c822f1d944c783de94bb044cce6f",
      "Inputs": [],
      "Outputs": [
        {
          "Id": "8ebf6faebf534b758aa4280687333d66",
          "Name": "Category",
          "Description": "The selected Category.",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Disabled",
      "Description": "All built-in categories."
    },
    {
      "ConcreteType": "DSRevitNodesUI.FamilyTypes, DSRevitNodesUI",
      "SelectedIndex": 2,
      "SelectedString": "A1 metric:A1 metric",
      "NodeType": "ExtensionNode",
      "Id": "d28aa0d4f68948ad882a26723e0d2d91",
      "Inputs": [],
      "Outputs": [
        {
          "Id": "0079b8afb62841b1932894c5ab5541c6",
          "Name": "Family Type",
          "Description": "The selected Family Type",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Disabled",
      "Description": "All family types available in the document."
    },
    {
      "ConcreteType": "DSRevitNodesUI.WallTypes, DSRevitNodesUI",
      "SelectedIndex": 0,
      "SelectedString": "Cavity wall_sliders",
      "NodeType": "ExtensionNode",
      "Id": "ad6d2c17fab44821a41ba6c5b66d7ce3",
      "Inputs": [],
      "Outputs": [
        {
          "Id": "6398879842bd4bddab84783f7aaddf10",
          "Name": "Wall Type",
          "Description": "The selected Wall Type",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Disabled",
      "Description": "All wall types available in the document."
    },
    {
      "ConcreteType": "DSRevitNodesUI.PerformanceAdviserRules, DSRevitNodesUI",
      "SelectedIndex": 0,
      "SelectedString": "Duplicate instances",
      "NodeType": "ExtensionNode",
      "Id": "535d483e53314764830eb56f5b6daf29",
      "Inputs": [],
      "Outputs": [
        {
          "Id": "8d7f4c5c81dc48b88d47bbee56d55809",
          "Name": "Performance Adviser Rules",
          "Description": "The selected Performance Adviser Rules",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Disabled",
      "Description": "All Performance Adviser rules."
    }
  ]
```

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@mjkkirschner

### CODE REVIEW

File: test/DynamoCoreTests/Nodes/DropDownTests.cs [hunk 1]
```diff
@@ -98,6 +113,31 @@ public void OpenXmlDYNwithSelectionNode()
...
         }
 
+        [Test]
+        public void GetSelectedStringFromItemShouldReturnString()
+        {
+            // Initialize the node
+            var node = new myDropDown();
+            node.PopulateItems();
+            //node.GetSelectedStringFromItem(node.Items.FirstOrDefault());
+            Assert.AreEqual("Monday", node.GetType().GetMethod("GetSelectedStringFromItem").Invoke(node, new object[] { node.Items.FirstOrDefault() }));
```
Comment (QilongTang) [inline test/DynamoCoreTests/Nodes/DropDownTests.cs]: Even after I exposed `GetSelectedStringFromItem`, I still could not invoke it. I checked the test setup, it should be fine. So this remains a mystery for me.. Now my machine refuse to run the test. Going to try my home machine.

File: test/DynamoCoreTests/Nodes/DropDownTests.cs [hunk 2]
```diff
@@ -98,6 +113,31 @@ public void OpenXmlDYNwithSelectionNode()
...
+            // Initialize the node
+            var node = new myDropDown();
+            node.PopulateItems();
+            node.SelectedIndex = 3;
+
+            // Populate the items again without modifying the items
+            node.PopulateItems();
+
+            // Index should stay the same
+            Assert.AreEqual(3, node.SelectedIndex);
```
Comment (QilongTang) [inline test/DynamoCoreTests/Nodes/DropDownTests.cs]: This test is to valid that calling `PopulateItems` should not change the original `SelectedIndex`

**Comment by QilongTang** [comment]
@mjkkirschner I commented out one test out so I can merge for now and cherry-pick, Revit team need this change to unblock the testing on their end I believe.

I will continue look at enbling that unit test

**Comment by mjkkirschner** [comment]
LGTM

**Pull request merged**